### PR TITLE
Refactor of using gophercloud

### PR DESCRIPTION
**Pull request by mrkisaolamb**

### CODE REVIEW

File: modules/openstack/openstack.go [hunk 1]
```diff
@@ -117,7 +114,6 @@ func NewOpenStack(
 	if err != nil {
 		return nil, err
 	}
-
```
Comment (gibizer) [inline modules/openstack/openstack.go:120]: nit: whitespace damage. Or is it go vet again? :)

File: modules/openstack/openstack.go [hunk 2]
```diff
@@ -61,17 +61,14 @@ type TLSConfig struct {
...
+func GetOpenStackProvider(
 	cfg AuthOpts,
-) (*OpenStack, error) {
-
+) (*gophercloud.ProviderClient, error) {
 	opts := gophercloud.AuthOptions{
 		IdentityEndpoint: cfg.AuthURL,
 		Username:         cfg.Username,
 		Password:         cfg.Password,
-		TenantName:       cfg.TenantName,
```
Comment (gibizer) [inline modules/openstack/openstack.go:74]: What is the reason you drop Tenant?

File: modules/test/apis/server.go [hunk 3]
```diff
@@ -18,11 +18,12 @@ import (
 	"net/http/httptest"
 
 	"github.com/go-logr/logr"
+	"github.com/gorilla/mux"
```
Comment (gibizer) [inline modules/test/apis/server.go:21]: can we make this in a separate commit? (can be in a same PR if you need it)

File: modules/openstack/openstack.go (hunk 1 above)
Comment (mrkisaolamb) [reply modules/openstack/openstack.go]: yes it was go vet

File: modules/openstack/openstack.go (hunk 2 above)
Comment (mrkisaolamb) [reply modules/openstack/openstack.go]: done

File: modules/test/apis/server.go (hunk 3 above)
Comment (mrkisaolamb) [reply modules/test/apis/server.go]: done

File: modules/openstack/openstack.go (hunk 1 above)
Comment (gibizer) [reply modules/openstack/openstack.go]: then ignore me and chalk up one more to the "random acts of govet"

**Review by gibizer — APPROVED** [review]
As far as I see this is a clean refactor of `NewOpenStack` so the users of NewOpenStack will not break.
